payjoin-cli/src/app/config.rs

```diff
@@ -204,105 +202,102 @@ impl Config {
 }
 
 /// Set up default values and CLI overrides for Bitcoin RPC connection settings
-fn add_bitcoind_defaults(builder: Builder, matches: &ArgMatches) -> Result<Builder, ConfigError> {
-    builder
+fn add_bitcoind_defaults(config: Builder, cli: &Cli) -> Result<Builder, ConfigError> {
+    // Set default values
+    let config = config
         .set_default("bitcoind.rpchost", "http://localhost:18443")?
-        .set_override_option(
-            "bitcoind.rpchost",
-            matches.get_one::<Url>("rpchost").map(|s| s.as_str()),
-        )?
         .set_default("bitcoind.cookie", None::<String>)?
-        .set_override_option(
-            "bitcoind.cookie",
-            matches.get_one::<String>("cookie_file").map(|s| s.as_str()),
-        )?
         .set_default("bitcoind.rpcuser", "bitcoin")?
-        .set_override_option(
-            "bitcoind.rpcuser",
-            matches.get_one::<String>("rpcuser").map(|s| s.as_str()),
-        )?
-        .set_default("bitcoind.rpcpassword", "")?
-        .set_override_option(
-            "bitcoind.rpcpassword",
-            matches.get_one::<String>("rpcpassword").map(|s| s.as_str()),
-        )
+        .set_default("bitcoind.rpcpassword", "")?;
```

Comment:
The missing defaults were what was causing the test failures before, so the tests caught a behavioral change I made. It didn't really make sense to me to provide defaults for these since `rpcuser` and `rpcpassword` don't have sensible defaults the way `rpchost` does, so it felt weird to give one here. I can take a look at refactoring these out of the tests later if we want.